### PR TITLE
Handle systemd-resolved, issue DNS commands via update-systemd-resolved

### PR DESCRIPTION
**Pull request by karlicoss**
TLDR: this PR is supposed to bring DNS leak protection fo systems managed by systemd-resolved.

Longer explanation: it seems that Ubuntu uses [systemd-resolved](https://www.freedesktop.org/software/systemd/man/systemd-resolved.service.html) now instead of `resolvconf` since 16.10. The way it works is this systemd resolver running on `127.0.0.53` and `/etc/resolv.conf` only has a single entry pointing at that local DNS server.

What was happening prior to that update: openpyn was detecting that `resolvconf` is not available (not sure when it was removed on Ubuntu, but unavailable on default `17.10`), and fallbacking to manual DNS patching, which works for DNS leak protection, but is not restoring `resolv.conf` to the original state.

What this patch does: `systemd-resolved` manages different DNS servers for different interfaces, e.g. default could be `8.8.8.8` and vpn (e.g. `tun0`) could use one of the NordVPN servers. One can set these servers via dbus communication with systemd-resolved, and on the `up/down` events in a way similar to `update-resolv-conf.sh` script, luckily this was [already implemented](https://github.com/jonathanio/update-systemd-resolved), and that's what I used.

After running openpyn with this patch, the `/etc/resolv.conf` is untouched and DNS servers are as follows:
```
$ systemd-resolve --status                                                                                                                             
Global
         DNS Servers: 8.8.8.8
         DNSSEC NTA: 10.in-addr.arpa
....
                      intranet
                      lan
                      local
                      private
                      test

Link 67 (tun0)
      Current Scopes: DNS
      ....
      DNS Servers: 103.86.96.100
                   103.86.99.100
```

I tested DNS resolving, looked at tcmdump and it appears as working, DNS requests go via nordvpn's DNS servers! After stopping VPN, everything works as before and goes via default (google in my case) DNS.

Some additional notes:

1. I rely on DNS servers pushed by NordVPN scripts rather than ones hardcoded in OpenPYN, do you think I should use hardcoded too? Also note that the ones picked up here are `103.86.96.100,103.86.99.100` (which agrees with [this](https://support.nordvpn.com/#/Other/1047409702/What-are-your-DNS-server-addresses.htm)), whereas you're using `162.242.211.137,78.46.223.24' (which agrees with [this](https://nordvpn.com/tutorials/d-link/pptp/)). Confusing and not sure which ones are right, hopefully all of them!
2. I could copy the `update-systemd-resolved` script in this repo (alike `update-resolv-conf.sh`), but not so sure about the license (it's GPL). Can we/should be just copy it, or it's fine prompting to install manually?
3. I've seen some pylint (and you've recently had something about pylama in commit history), but I didn't find any pylint configs, and default `pylama openpyn` command spits out warnings all over the project, so I assume you are running it with some flags? Let me know how are you running pylama and I'll fix the codestyle.

Let me know what do you think! Thanks.

### CODE REVIEW

**Comment by yuriw** [comment]
@karlicoss I have my firewall rules preventing any client on my network to use external DNS server and using only my router IP as DNS. Will this solution support that ?

**Comment by karlicoss** [comment]
@yuriw so if your router is handling DNS, you must have been using openpyn with `--skip-dns-patch` option? In that case nothing would change for you. Otherwise, please elaborate.

**Comment by jotyGill** [comment]
Hey karlicoss, thank you for your contribution. Could you please send this pull request in my "test" branch. I only (try to) push commits to master branch after some testing.

**Comment by jotyGill** [comment]
@yuriw where are these firewall rules applied? on your router? if so , It shouldn't matter as your router will only see encrypted traffic of the vpn tunnel, It can't tell whats dns traffic whats not.

**Comment by yuriw** [comment]
@jotyGill that’s a good point, I never thought about this that way, thx !

**Comment by 1951FDG** [comment]
@karlicoss Hi, regarding bullet point 3, pylint and pylama, this link can help you, it contains the configs, https://github.com/1951FDG/openpyn-nordvpn/pull/7#issuecomment-375151853

**Comment by karlicoss** [comment]
@jotyGill Hiya, I changed the base branch to 'test'. It's also showing two of your merge commits, but I guess they would disappear once you mege master back to test. Or shall I branch off the test?

**Comment by karlicoss** [comment]
@jotyGill I also emailed NordVPN support today regarding DNS servers, and they replied:

```
Hello, Dmitrii,Thank you for your letter. Our correct DNS addresses are 103.86.96.100 and 103.86.99.100.The https://nordvpn.com/tutorials/ page has outdated tutorials and you should use https://support.nordvpn.com/ instead. Let us know if you need any further assistance.
```

So I guess we should switch the hardcoded ones to `103.86.96.100` and `103.86.99.100` !

**Comment by 1951FDG** [comment]
@karlicoss I also noticed this today, but the old ones still work, and are actually faster than the new ones, but yeah, I guess we should switch the hardcoded ones to `103.86.99.100` and `103.86.96.100` if for when the old DNS servers expire !

**Comment by jotyGill** [comment]
I have added the https://github.com/jonathanio/update-systemd-resolved in the scripts folder. That's the beauty of GPL, you can use and modify, do what ever you want :)
Same goes for my script here (openpyn).
